targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_LPC54114/device/TARGET_LPC54114_M4/TOOLCHAIN_ARM_STD/LPC54114J256_cm4.sct

```diff
@@ -121,6 +121,8 @@ LR_IROM1 m_interrupts_start m_text_start+m_text_size-m_interrupts_start { ; load
   }
   RW_IRAM1 ((ImageLimit(RW_m_data) == m_data_start) ? ImageLimit(RW_m_data) : +0) EMPTY Heap_Size { ; Heap region growing up
   }
+  ARM_LIB_HEAP AlignExpr(+0, 16) EMPTY Heap_Size{ ; Heap region growing up
```

Comment:
`Heap_Size` should probably be calculated as done for other platforms (`MBED_RAM_SIZE - RAM_FIXED_SIZE - (AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_RAM1_START)`) instead of setting it to `0x0400`.
@ARMmbed/team-nxp can you confirm?

Reply:
@hugueskamba I've had to remove the dynamic calculation of the heap size since it was causing CI to fail. The change I made resulted in a circular definition between Heap_Size and RW_IRAM1.

Reply:
Can you show me what you had that caused the circular definition?

Reply:
This was the original change:
```c++
#if (defined(__heap_size__))
  #define Heap_Size                    __heap_size__
#else
  #define Heap_Size                    (m_data_size - Stack_Size - (AlignExpr(ImageLimit(RW_IRAM1), 16) - m_data_start))
#endif
```

Reply:
@ARMmbed/team-nxp any thoughts?